packages/code-studio/src/settings/FormattingSectionContent.tsx

```diff
@@ -505,14 +623,15 @@ export class FormattingSectionContent extends PureComponent {
       truncateNumbersWithPound,
     } = this.state;
 
-    const formatter = formatSettings
-      .filter(FormattingSectionContent.isFormatRuleValidForSave)
-      .map(FormattingSectionContent.removeFormatRuleExtraProps);
+    const formatter =
+      formatSettings
+        ?.filter(FormattingSectionContent.isFormatRuleValidForSave)
+        .map(FormattingSectionContent.removeFormatRuleExtraProps) ?? [];
 
     const { settings, saveSettings } = this.props;
-    const newSettings = {
+    const newSettings: Partial<WorkspaceSettings> = {
```

Comment:
Should just be `WorkspaceSettings`, not `Partial`.
```suggestion
    const newSettings: WorkspaceSettings = {
```